Core and Platforms - Updated for latest libnx and fixed compile error

EmmmaTech opened the pull request:
This is identical to #164 except an additional compile error was fixed.

EmmmaTech committed:
Fixed build errors and updated for newest libnx version

DarkMatterCore commented:
(for the record, my PR is for the legacy branch)

natinusala commented:
Thanks for the PR! Why change the optional include? I'm not fan of changing 
external code like that

Le 11 août 2021 09:53:07 Pablo Curiel ***@***.***> a écrit :
>
> (for the record, my PR is for the legacy branch)
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android.

EmmmaTech commented:
I think I took it from somewhere else, I'll revert the change and put the optional header on top of the <nanovg/dk_renderer.hpp> header.

EmmmaTech commented:
Actually, that doesn't fix it. If you compile the code, you'll get this error:
```bash
In file included from /Users/emreterzioglu/Documents/C++_Programs/borealis-fork/working-yoga/./library/lib/extern/nanovg-deko3d/source/dk_renderer.cpp:1:
/Users/emreterzioglu/Documents/C++_Programs/borealis-fork/working-yoga/./library/lib/extern/nanovg-deko3d/include/nanovg/dk_renderer.hpp:168:18: 
error: 'optional' in namespace 'std' does not name a template type
  168 |             std::optional<CMemPool::Handle> m_vertex_buffer;
      |                  ^~~~~~~~
```
This means the optional include has to be moved into the `dk_renderer.hpp` file.

EmmmaTech committed:
Temp workaround to "optional" include header error

EmmmaTech commented:
I decided to open an issue in the `nanovg-deko3d` repo about this compiler error.

EmmmaTech commented:
Alright, so a fix was merged into `nanovg-deko3d`, so I can pull the change

EmmmaTech committed:
git subrepo pull (merge) library/lib/extern/nanovg-deko3d

subrepo:
  subdir:   "library/lib/extern/nanovg-deko3d"
  merged:   "5b8661b"
upstream:
  origin:   "https://github.com/Adubbz/nanovg-deko3d"
  branch:   "master"
  commit:   "bb69eb5"
git-subrepo:
  version:  "0.4.3"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "2f68596"

EmmmaTech committed:
Few changes

JSH32 commented:
Any plans to merge the fix? Compile error still present and I'm not able to include as a submodule because of it

EmmmaTech commented:
I'm awaiting review from @natinusala 
They have been inactive in this repo, so PRs aren't really being reviewed.
